tests/test_graph_metrics.py

```diff
@@ -280,13 +280,18 @@ def test_metrics(dialect, test_helpers):
 
     df_nm = cm.nodes.as_pandas_dataframe()
 
-    for unique_id, expected_node_degree in expected_node_degrees:
+    for unique_id, expected_degree, expected_centrality in expected_node_metrics:
         relevant_row = df_nm[df_nm["composite_unique_id"] == unique_id]
         calculated_node_degree = relevant_row["node_degree"].iloc[0]
-        assert calculated_node_degree == expected_node_degree, (
-            f"Expected node degree {expected_node_degree} for node {unique_id}, "
+        assert calculated_node_degree == expected_degree, (
+            f"Expected node degree {expected_degree} for node {unique_id}, "
             f"but found node degree {calculated_node_degree}"
         )
+        calculated_node_centrality = relevant_row["node_centrality"].iloc[0]
+        assert round(calculated_node_centrality, 3) == round(expected_centrality, 3), (
```

Comment:
re: test failures you might need to cast to `float` to avoid the type mismatch - see the density + cluster centralisation tests for comparison